evap/staff/importers.py

```diff
@@ -97,28 +98,85 @@ def store_in_database(self, vote_start_datetime, vote_end_date, semester):
+                return True
+            if name_collision == "both":
+                importer.errors[ImporterError.COURSE].append(
+                    _("Course {}({}) does already exist in this semester.").format(self.name_en, self.name_de)
+                )
+            else:
+                importer.errors[ImporterError.COURSE].append(
+                    _("Course {} does already exist in this semester.").format(
+                        self.name_en if name_collision == "name_en" else self.name_de
+                    )
+                )
+        return False
+
+    def check_name_collision(self, semester):
+        name_collision = None
+        if Course.objects.filter(semester=semester, name_en=self.name_en).exists():
+            name_collision = "name_en"
+        if Course.objects.filter(semester=semester, name_de=self.name_de).exists():
+            name_collision = "name_de" if not name_collision else "both"
+        return name_collision
+
+    def find_and_set_existing_course(self, semester, importer):
+        courses = Course.objects.filter(
+            semester=semester,
             name_en=self.name_en,
+            name_de=self.name_de,
             type=self.course_type,
-            semester=semester,
-        )
-        course.save()
-        course.responsibles.set([responsible_dbobj])
-        course.degrees.set(self.degrees)
-        evaluation = Evaluation(
-            vote_start_datetime=vote_start_datetime,
-            vote_end_date=vote_end_date,
-            course=course,
-            wait_for_grade_upload_before_publishing=self.is_graded,
-        )
-        evaluation.save()
-        evaluation.contributions.create(
-            evaluation=evaluation,
-            contributor=responsible_dbobj,
-            role=Contribution.Role.EDITOR,
-            textanswer_visibility=Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS,
+            responsibles__email=self.responsible_email,
         )
+        if courses.exists():
+            course = courses.first()
+            evaluations = Evaluation.objects.filter(course=course)
+            if len(evaluations) == 1:
+                if evaluations.first().wait_for_grade_upload_before_publishing == self.is_graded:
+                    self.existing_course = course
+                    return course
+                importer.errors[ImporterError.COURSE].append(
+                    _("Course {}({}) does already exist in this semester but the only evaluation is {}graded.").format(
+                        self.name_en, self.name_de, "not " if self.is_graded else ""
+                    )
+                )
+            else:
+                importer.errors[ImporterError.COURSE].append(
+                    _(
+                        "Course {}({}) does already exist in this semester and is identical but has {} evaluations instead of 1."
+                    ).format(self.name_en, self.name_de, len(evaluations))
+                )
```

Comment:
Do you think these error messages are too detailed? Alternatively there could be a general error that just states that the course is similar to one that exists but differs in some attributes that must be equal. With this, these two error messages could also be merged into one

Reply:
The error messages for differing attributes should list which attributes are different but don't need to include details about the values.
An explicit message for courses that already have more than one evaluation is also needed to explain this different kind of obstacle.

Reply:
Ok, sounds good. I dont know if its too much but now I included another error message if the course type or responsible person dont match